app/code/Magento/CatalogImportExport/Model/Export/Product.php

```diff
@@ -949,8 +949,9 @@ protected function getExportData()
             $productIds = array_keys($rawData);
             $stockItemRows = $this->prepareCatalogInventory($productIds);
 
+            $entityCollection = clone $this->_getEntityCollection();
```

Comment:
I believe such change is backward-incompatible, by doing this we are delegating the responsibility of preparing the collection to any method that calls `getExportData`, which is true for our implementations, but since this is an @api it may not be true for partners and other customers. Such change would change their application behavior and may introduce bugs.

I've thought about making the collection a parameter for `getExportData` but that is also backward-incompatible, can we some how check if the collection is prepared?